addons/stock/report/stock_forecasted.py

```diff
@@ -133,7 +133,15 @@ def _get_report_data(self, product_template_ids=False, product_variant_ids=False
         res = {}
 
         if self.env.context.get('warehouse'):
-            warehouse = self.env['stock.warehouse'].browse(self.env.context.get('warehouse'))
+            if isinstance(self.env.context['warehouse'], int):
+                warehouse = self.env['stock.warehouse'].browse(self.env.context['warehouse'])
+            else:
+                warehouse = self.env['stock.warehouse'].search(
+                    ['|',
+                        ('name', 'ilike', self.env.context['warehouse'][0]),
+                        ('code', 'ilike', self.env.context['warehouse'][0]),
+                    ], limit=1
+                )
```

Comment:
I don't understand what happened, but now when I test (in your latest runbot) the report now correctly opens with the filtered warehouse :scream_cat: So I guess you fixed it?

Anyways, the `sale_stock_renting` seems to be a separate bug, or there's something weird/inconsistent going on with how the warehouse filter is being passed into the context.

I'm not sure what you prefer to do, investigate more and see if you can fix it for both at the same time or just r+ this PR? I'll leave it to you to decide. Let me know if you need me to test some more after doing changes.

Reply:
> I don't understand what happened, but now when I test (in your latest runbot) the report now correctly opens with the filtered warehouse 🙀 So I guess you fixed it?

I just applied your suggestion, so you're the wizard here.

> Anyways, the `sale_stock_renting` seems to be a separate bug, or there's something weird/inconsistent going on with how the warehouse filter is being passed into the context.
> 
> I'm not sure what you prefer to do, investigate more and see if you can fix it for both at the same time or just r+ this PR? I'll leave it to you to decide. Let me know if you need me to test some more after doing changes.

I'll test the fwp of this PR in all versions. Since I will probably encounter a runbot/cherry-pick error in master/saas~17.3, I'll create an enterprise branch there and (try to) fix there.